alma/controllers/hook/displayShoppingCartFooter.php

```diff
@@ -39,30 +39,44 @@ public function run($params)
             return null;
         }
 
-        $eligibility = AlmaEligibilityHelper::eligibilityCheck($this->context);
-        $eligibilityMsg = AlmaSettings::getEligibilityMessage();
-
-        if (!$eligibility->isEligible) {
-            $eligibilityMsg = AlmaSettings::getNonEligibilityMessage();
-
+        $eligibilities = AlmaEligibilityHelper::eligibilityCheck($this->context);
+        $eligible = false;
+        foreach($eligibilities as $eligibility){
+            if($eligibility->isEligible){
+                $eligible = true;
+            }
+        }        
+        if(!$eligible){
             $cart = $this->context->cart;
             $cartTotal = almaPriceToCents((float) $cart->getOrderTotal(true, Cart::BOTH));
-            $minAmount = $eligibility->constraints['purchase_amount']['minimum'];
-            $maxAmount = $eligibility->constraints['purchase_amount']['maximum'];
-
-            if ($cartTotal < $minAmount || $cartTotal > $maxAmount) {
-                if ($cartTotal > $maxAmount) {
-                    $eligibilityMsg .= ' ' . sprintf(
-                        $this->module->l('(Maximum amount: %s)', 'displayShoppingCartFooter'),
-                        Tools::displayPrice(almaPriceFromCents($maxAmount))
-                    );
-                } else {
-                    $eligibilityMsg .= ' ' . sprintf(
-                        $this->module->l('(Minimum amount: %s)', 'displayShoppingCartFooter'),
-                        Tools::displayPrice(almaPriceFromCents($minAmount))
-                    );
-                }
+            $minimum = 9999999;
+            $maximum = 0;
+            foreach($eligibilities as $eligibility){
+                if(!$eligibility->isEligible){
+                    $minAmount = $eligibility->constraints['purchase_amount']['minimum'];
+                    $maxAmount = $eligibility->constraints['purchase_amount']['maximum'];
+                    if ($cartTotal < $minAmount || $cartTotal > $maxAmount) {
+                        if ($cartTotal > $maxAmount && $maxAmount > $maximum) {
+                            $eligibilityMsg = ' ' . sprintf(
+                                $this->module->l('(Maximum amount: %s)', 'displayShoppingCartFooter'),
+                                Tools::displayPrice(almaPriceFromCents($maxAmount))
+                            );
+                            $maximum = $maxAmount;
+                        }
+                        if ($cartTotal < $minAmount && $minAmount < $minimum) {
+                            $eligibilityMsg = ' ' . sprintf(
```

Comment:
Use the loop only to identify the global min/max values, then compute the message string after the loop depending on whether `$cartTotal` is higher than global max or lower than global min